### PR TITLE
[FE-15602] Popup image support

### PR DESCRIPTION
**Pull request by johallar**
This adds support for showing images in the popup for raster layers. This includes:

* Adding the actual image tag in the popup html + styles
* Safety check for image size, if > 5MB it will just show the link instead of image preview
* Async-ifies the whole popup html generation process, allows us to check image size
* Loader for popups which have to do async things (images only RN). Loader only shows up if making the requests takes > half a second so we don't flash it in front of the user for every popup
* Copy popup functionality copies the image url when an image preview is displayed

Dataset is on Kali @ `inaturalist_observations_colorado_fungi`

Basic demo + slow connection mocked with chrome to show loader behavior:

https://github.com/heavyai/heavyai-charting/assets/1031659/24c1d8bc-8fbd-4e10-86f2-2d251eb07402

copying from the popup with a tooltip gives you the url of the image:
Screen Shot 2023-07-12 at 2 19 32 PM

```
species_guess: Xanthoparmelia
image_url: https://inaturalist-open-data.s3.amazonaws.com/photos/595549/medium.JPG
created_at: 12/2/2013, 4:42:49 PM
num_identification_agreements: 1.00
place_state_name: Colorado
user_id: 26K
url: http://www.inaturalist.org/observations/472419
```

### CODE REVIEW

**Review by cmatzenbach — CHANGES_REQUESTED** [review]
Only one area where changes needed - everything else works well and code looks good

File: src/mixins/raster-layer.js [hunk 1]
```diff
@@ -392,63 +398,171 @@ export default function rasterLayer(layerType) {
...
+  function filenameHasExtension(filename, extensions) {
+    return extensions.some(ext => filename.toLowerCase().endsWith(ext))
+  }
+
+  const LinkElement = (href, content) =>
+    `<a href="${href}" target="_blank" rel="noopener noreferrer">
+    ${content}
+    </a>`
+
+  const imageExtensions = [
```
[the remark this answers is not in the document]
Comment (johallar) [reply src/mixins/raster-layer.js]: 👍 Changes made whenever you can take another look

**Comment by johallar** [comment]
I'm going to make one update here to handle query string parameters w image previews. Current check is a little too dumb `endsWith`, and sounds like some use cases will need to be able to parse image names with query string params after.

**Comment by johallar** [comment]
OK @cmatzenbach, made an update to parse URLs a little smarter so we can handle query string params and more. At your earliest convenience

**Review by cmatzenbach — APPROVED** [review]
LGTM

File: src/mixins/raster-layer.js [hunk 2]
```diff
@@ -392,63 +399,161 @@ export default function rasterLayer(layerType) {
     return _layer._areResultsValidForPopup(results[0])
   }
 
-  function replaceURL(colVal) {
-    if (typeof colVal === "string") {
-      const urlRegExpr = /(((https?:\/\/)|(www\.))[^\s^<>'"”`]+)/g
-      return colVal.replace(urlRegExpr, url => {
+  function filenameHasExtension(filename, extensions) {
```
[the remark this answers is not in the document]
Comment (johallar) [reply src/mixins/raster-layer.js]: Fixed this up [here](https://github.com/heavyai/heavyai-charting/pull/650/files#diff-17533c14138ae3b018fdbf3302ac845357e33c691349841654b30d2566e6be24R453)

**Pull request merged**